### PR TITLE
part-2-testing

### PR DESCRIPTION
**Pull request by AdielDror**
I added tests

### CODE REVIEW

File: chess_gui.py [hunk 1]
```diff
@@ -156,9 +159,11 @@ def main():
                             if human_player is 'w':
                                 ai_move = ai.minimax_white(game_state, 3, -100000, 100000, True, Player.PLAYER_2)
                                 game_state.move_piece(ai_move[0], ai_move[1], True)
+                                logger.info("The white player started first.")
```
Comment (talbarfm) [inline chess_gui.py:162]: The question is if the computer started or the player

File: chess_gui.py [hunk 2]
```diff
@@ -181,12 +186,16 @@ def main():
         if endgame == 0:
             game_over = True
             draw_text(screen, "Black wins.")
+            logger.info("Black is wins!")
```
Comment (talbarfm) [inline chess_gui.py:189]: Add more logs to the endgame

**Review by talbarfm — CHANGES_REQUESTED** [review]
Please note that you have to separate the pull requests.

File: Tests/test_4Piece.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    game_state_1 = game_state()
+    white_knight_1 = Knight('n', 3, 4, Player.PLAYER_1)
+    game_state_1.board = [[(), (), (), (), (), (), (), ()],
+                          [(), (), (), (), (), (), (), ()],
+                          [(), (), (), (), (), (), (), ()],
+                          [(), (), (), (), white_knight_1, (), (), ()],
+                          [(), (), (), (), (), (), (), ()],
+                          [(), (), (), (), (), (), (), ()],
+                          [(), (), (), (), (), (), (), ()],
+                          [(), (), (), (), (), (), (), ()]]
```
Comment (talbarfm) [inline Tests/test_4Piece.py]: Use mocks, or create an initialization function

File: Tests/test_4Piece.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+    # Initialize Black Pieces
+    black_rook_1 = Rook('r', 7, 0, Player.PLAYER_2)
+    black_rook_2 = Rook('r', 7, 7, Player.PLAYER_2)
+    black_knight_1 = Knight('n', 7, 1, Player.PLAYER_2)
+    black_knight_2 = Knight('n', 7, 6, Player.PLAYER_2)
+    black_bishop_1 = Bishop('b', 7, 2, Player.PLAYER_2)
+    black_bishop_2 = Bishop('b', 7, 5, Player.PLAYER_2)
+    black_queen = Queen('q', 7, 4, Player.PLAYER_2)
+    black_king = King('k', 7, 3, Player.PLAYER_2)
```
Comment (talbarfm) [inline Tests/test_4Piece.py]: There's a better way to initialize the game

File: Tests/test_4Piece.py (hunk 3 above)
Comment (AdielDror) [reply Tests/test_4Piece.py]: I fixed it

File: Tests/test_4Piece.py (hunk 4 above)
Comment (AdielDror) [reply Tests/test_4Piece.py]: I fixed it.

**Pull request merged**